### ISSUE TITLE
It works for a while then closes by itself

### ISSUE THREAD

**Issue by lit377** [ogkalu2/comic-translate#194]
i tried to debug it but i'm no expert, not at all. also i'm not sure if it has its own error log or not, if it does place let me know where i can find it, i did find two files "errors.py" and "logs.py".

here's the error i got from my own error logs i created for the app.

2024-12-29 18:27:02,797 - DEBUG - matplotlib data path: D:\comic-translate-main\comic-translate-main\venv\lib\site-packages\matplotlib\mpl-data
2024-12-29 18:27:02,802 - DEBUG - CONFIGDIR=C:\Users\x\.matplotlib
2024-12-29 18:27:02,804 - DEBUG - interactive is False
2024-12-29 18:27:02,804 - DEBUG - platform is win32
2024-12-29 18:27:02,871 - DEBUG - CACHEDIR=C:\Users\x\.matplotlib
2024-12-29 18:27:02,874 - DEBUG - Using fontManager instance from C:\Users\x\.matplotlib\fontlist-v390.json
2024-12-29 18:27:04,559 - INFO - pikepdf C++ to Python logger bridge initialized
2024-12-29 18:27:09,188 - ERROR - An error occurred
Traceback (most recent call last):
  File "D:\comic-translate-main\comic-translate-main\comic.py", line 1387, in <module>
    app.setWindowIcon(QIcon(icon_resource.get_icon_path()))
AttributeError: '_GeneratorContextManager' object has no attribute 'setWindowIcon'

i'm thinking of reinstalling it through cloning method. 

but all in all, it's one of the most proper tool for translation i have found till now, it works pretty well for text segmentation, kudos to the creators! 

and btw i'm running it in a virtual environment.

**Comment by ogkalu2** [comment]
Should be fixed now

**Comment by lit377** [comment]
it's working goood! if possible i want to improve the quality of translation it produces, by adding additional features like saving/adding glossary, tone setting, era appropriate slangs and idioms, saving the name of characters and terminology, keeping things consistence across the chapters, each project file to have its own history or general database, like a config file. i'm not so sure what specific files i'll have to edit to implement all these additions so if possible please do guide this noob! i tried integrating deepseek v3 cause it's really cheap and really good, but kinda failed? i'm really enjoying using comic-translate 🐱‍🏍

**Comment by ogkalu2** [comment]
In Settings>LLMs, you can write whatever information for the project you need the LLM to know, but it's not saved for each project (I'll add support for that). 

History of what ?

**Comment by lit377** [comment]
yes, that feature is very handy but it can't save the glossary for each series, i'm trying to add the features i find the need of every day, but i'm not good at yet 😭. thinking of making a database of terms curated to specific genres, i know it'll not be useful in the future as llm gets smarter but for now it needs the data is what i think. i'm thinking about it saving important information like relationships, items etc as history and save it in the ctpr file if possible, thinking of going with css or json for data base where it can retrieve the saved information. i made a beta glossary manager keeping my kind of requirements in mind, yeah i know i can just keep a separate glossary and use it each time, but i'm good at being lazy! 
![comix3proof](https://github.com/user-attachments/assets/82f955ae-123c-4e2e-9836-c6f5c0314064)

**Comment by lit377** [comment]
i think i should move this to discussion and close the issue ticket

**Comment by ogkalu2** [comment]
Yes, move this to discussions or open another issue as a feature request. You can the save the glossary in the ctpr file but get the beta working first. UI wise, might be best to move this to be under LLMs as other kinds of translators wouldn't be able to utilize any of this. 

A separate ui for term/character glossaries or slangs makes sense but i'm not sure i see the sense in separate uis for anything else (complex relationship info can just be written in the general box)

**Comment by ogkalu2** [comment]
I mean Glossary stuff should probably be inside the LLM tab rather than its own tab
